### ISSUE TITLE
Refactor network class interface

### ISSUE THREAD

**Issue by tfujiwar** [blue-oil/blueoil#961]
This issue was proposed before but there is no progress.
https://github.com/blue-oil/blueoil/pull/548#issuecomment-545253771

## Problem
Currently, network class methods (e.g. `placeholders`, `inference`, `loss`, etc.) creates new operators when we call them. When we train networks, we need to construct a graph in training scripts by using those methods like this. But ideally, network classes should be responsible to construct a graph.
```py
# Create placeholders
is_training_placeholder = tf.compat.v1.placeholder(tf.bool, name="is_training_placeholder")
images_placeholder, labels_placeholder = model.placeholders()

# Create train op
output = model.inference(images_placeholder, is_training_placeholder)
loss = model.loss(output, labels_placeholder)
opt = model.optimizer(global_step)
train_op = model.train(loss, opt, global_step)

# Create metrics op
metrics_ops_dict, metrics_update_op = model.metrics(output, labels_placeholder)
model.summary(output, labels_placeholder)
summary_op = tf.compat.v1.summary.merge_all()
metrics_summary_op = executor.metrics_summary_op(metrics_ops_dict)
```

Because of this, we need many local variables in a training script and it makes our training loop unnecessarily complicated.

## Solution
Initialize those operators (placeholders, train_op, summary_op) on constructing a network class instance and store them as properties of the instance.

networks/base.py
```py
class BaseNetwork(object):
    def __init__(...):
        ...
        self.images, self.labels = self.placeholder()
        self.is_training = tf.compat.v1.placeholder(tf.bool, name="is_training_placeholder")
        self.global_step = tf.Variable(0, trainable=False)

        output = self.inference(self.images, self.is_training)
        loss = self.loss(output, self.labels)
        optimizer = self.optimizer(global_step)

        self.train_op = self.train(loss, optimizer, global_step)
```

## TODO
- [x] Move global_step to network classes
- [ ] Move is_training_placeholder to network classes
- [ ] Store placeholders as properties
- [ ] Store train_op as a property
- [ ] Store summary_op as a property
- [ ] Store metrics_op as a property
